### ISSUE TITLE
Control EP Buffer addresses need to be manually set for PIC16 devices

### ISSUE THREAD

**Issue by infinity-computers-dot-net** [johnnydrazzi/USB-Stack#3]
What does the `control EP Buffer addresses need to be manually set for PIC16 devices` warning mean, and what should I do about it?

I'm trying to use a PIC16F1454 with the Serial UART example.

**Comment by johnnydrazzi** [comment]
The PIC16 devices have banked RAM and some of that RAM can be shared with the USB peripheral (0x2000 - 0x21EF). When the USB peripheral is enabled, some of your RAM become essentially SFRs (special function registers). These are the Buffer Descriptor Tables. The BDT's are used to control the buffers used by the USB peripheral to store incoming and outgoing EP data. The BDnADRL & BDnADRH contain the memory location to where you want data for a corresponding EP will be placed. 

Because the PIC16 devices have banked memory, one has to be careful not to place buffers in locations that will cross a bank boundary (see page 26 of the PIC16F145X datasheet). For PIC18 this isn't an issue, so it is convenient to place the EP buffers sequentially one after the other.

In usb.h line 100 onward, I've manually set the locations for the control EP buffer (EP0) for PIC16 devices. You shouldn't need to modify this for your own projects, but I've placed a warning there to inform the user. The same goes for EP1 in usb_cdc.h, usb_hid.h and usb_msd.h... All the examples have been thoroughly tested and don't need modification.
